### PR TITLE
Fix platform verification

### PR DESCRIPTION
**Pull request by stephanos**
## What was changed

Checking the binaries without _running_ the Docker container.

Example output:
```
verifying binaries in '/usr/local/bin' from 'temporaliotest/admin-tools:sha-05ac5b8' for platform 'linux/arm64' are 'ARM'

>  docker create --name bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ --platform linux/arm64 temporaliotest/admin-tools:sha-05ac5b8
Unable to find image 'temporaliotest/admin-tools:sha-05ac5b8' locally
sha-05ac5b8: Pulling from temporaliotest/admin-tools
Digest: sha256:3dc1f7ae18a0382d57cd6f75d69bfba4cce7b2f31199a9d097c411afc7b38c7d
Status: Downloaded newer image for temporaliotest/admin-tools:sha-05ac5b8
e22bfda97164763452dfbac597767d427e434571cd1c69598cb943259d5a7373


>  docker cp bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ:/usr/local/bin/. /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ2816749683


>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ2816749683/tctl
tctl is ARM

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ2816749683/tctl-authorization-plugin
tctl-authorization-plugin is ARM

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ2816749683/tdbg
tdbg is ARM

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ2816749683/temporal
temporal is ARM

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ2816749683/temporal-cassandra-tool
temporal-cassandra-tool is ARM

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYXJtNjQtMTcyMDcxOTk3MDU0OQ2816749683/temporal-sql-tool
temporal-sql-tool is ARM

verifying binaries in '/usr/local/bin' from 'temporaliotest/admin-tools:sha-05ac5b8' for platform 'linux/amd64' are 'x86'

>  docker create --name bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ --platform linux/amd64 temporaliotest/admin-tools:sha-05ac5b8
Unable to find image 'temporaliotest/admin-tools:sha-05ac5b8' locally
sha-05ac5b8: Pulling from temporaliotest/admin-tools
Digest: sha256:3dc1f7ae18a0382d57cd6f75d69bfba4cce7b2f31199a9d097c411afc7b38c7d
Status: Downloaded newer image for temporaliotest/admin-tools:sha-05ac5b8
d94c61fee08f7363197daa5fb2d73fba019857592754aa5489fde40f601ad5a7


>  docker cp bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ:/usr/local/bin/. /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ233242614


>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ233242614/tctl
tctl is x86

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ233242614/tctl-authorization-plugin
tctl-authorization-plugin is x86

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ233242614/tdbg
tdbg is x86

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ233242614/temporal
temporal is x86

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ233242614/temporal-cassandra-tool
temporal-cassandra-tool is x86

>  file /var/folders/4w/5qdjw8sd6417nldg5pvhs_rr0000gn/T/bGludXgvYW1kNjQtMTcyMDcxOTk3MjkzOQ233242614/temporal-sql-tool
temporal-sql-tool is x86
```

## Why?

Running the Docker container requires more setup for separate x86 and ARM environments. This way is easier.

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: src/image_copy/main.go [hunk 1]
```diff
@@ -69,44 +73,59 @@ func getTags(dstTag string, updateMajor bool) []string {
...
+	return string(out)
+}
+
 func mustHaveArchitecture(imageTag, platform, expectedArch string) {
-	fmt.Printf("verifying binaries in '%s' from '%s' for platform '%s' are '%s'\n", binFolder, imageTag, platform, expectedArch)
+	fmt.Printf("\nverifying binaries in '%s' from '%s' for platform '%s' are '%s'\n", binFolder, imageTag, platform, expectedArch)
 
-	cmd := exec.Command(
+	// create local container from image
+	containerId := base64.RawURLEncoding.EncodeToString([]byte(fmt.Sprintf("%v-%v", platform, time.Now().UnixMilli())))
```
Comment (dnr) [inline src/image_copy/main.go:91]: could just be a random string?

File: src/image_copy/main.go [hunk 2]
```diff
@@ -69,44 +73,59 @@ func getTags(dstTag string, updateMajor bool) []string {
 	return tags
 }
 
+func execCmd(name string, args ...string) string {
+	cmd := exec.Command(name, args...)
+	fmt.Println("\n> ", name, strings.Join(args, " "))
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		fmt.Println(out)
```
Comment (dnr) [inline src/image_copy/main.go:81]: do you need a string() here? sometimes go likes to dump []byte as a list of integers

File: src/image_copy/main.go [hunk 3]
```diff
@@ -69,44 +73,59 @@ func getTags(dstTag string, updateMajor bool) []string {
...
 	if err != nil {
 		log.Fatal(err.Error())
 	}
+	if len(files) == 0 {
+		panic("no binaries were found")
+	}
+	for _, file := range files {
+		out = execCmd("file", filepath.Join(tmpDir, file.Name()))
+		if !strings.Contains(out, expectedArch) {
+			panic(file.Name() + " is NOT " + expectedArch)
```
Comment (dnr) [inline src/image_copy/main.go:124]: log.Fatal?

File: src/image_copy/main.go [hunk 4]
```diff
@@ -69,44 +73,59 @@ func getTags(dstTag string, updateMajor bool) []string {
...
+		tmpDir)
+	fmt.Println(out)
+
+	// verify binaries
+	files, err := ioutil.ReadDir(tmpDir)
 	if err != nil {
 		log.Fatal(err.Error())
 	}
+	if len(files) == 0 {
+		panic("no binaries were found")
```
Comment (dnr) [inline src/image_copy/main.go:119]: log.Fatal?

File: src/image_copy/main.go [hunk 5]
```diff
@@ -69,44 +73,59 @@ func getTags(dstTag string, updateMajor bool) []string {
...
+	}
+	for _, file := range files {
+		out = execCmd("file", filepath.Join(tmpDir, file.Name()))
+		if !strings.Contains(out, expectedArch) {
```
Comment (dnr) [inline src/image_copy/main.go:123]: this would break if we put a shell script in there.. maybe let's worry about it if it happens?

**Review by dnr — APPROVED** [review]

File: src/image_copy/main.go [hunk 6]
```diff
@@ -69,44 +73,59 @@ func getTags(dstTag string, updateMajor bool) []string {
...
-
-if [ $count -lt 1 ]; then
-  echo "no binaries were found"
-  exit 1
-fi
-`)
-	out, err := cmd.CombinedOutput()
-	fmt.Println(string(out))
+		imageTag)
+	fmt.Println(out)
```
Comment (dnr) [inline src/image_copy/main.go:98]: you could also defer docker rm containerId to clean up better

File: src/image_copy/main.go (hunk 1 above)
Comment (stephanos) [reply src/image_copy/main.go:91]: Right; good catch!

File: src/image_copy/main.go (hunk 5 above)
Comment (stephanos) [reply src/image_copy/main.go:123]: True. Let's!

File: src/image_copy/main.go (hunk 2 above)
Comment (stephanos) [reply src/image_copy/main.go:81]: Right; good catch!

File: src/image_copy/main.go (hunk 1 above)
Comment (stephanos) [reply src/image_copy/main.go:91]: This was my way of generating a random string without a lot of ceremony. Is there a shorter/better way?

File: src/image_copy/main.go (hunk 6 above)
Comment (stephanos) [reply src/image_copy/main.go:98]: Added!

File: src/image_copy/main.go (hunk 1 above)
Comment (dnr) [reply src/image_copy/main.go:91]: `uuid.NewString()`?

File: src/image_copy/main.go [hunk 7]
```diff
@@ -69,44 +73,60 @@ func getTags(dstTag string, updateMajor bool) []string {
...
+	defer os.RemoveAll(tmpDir)
+	out = execCmd(
+		"docker",
+		"cp",
+		containerId+":"+binFolder+"/.",
+		tmpDir)
+	fmt.Println(out)
+
+	// verify binaries
+	files, err := ioutil.ReadDir(tmpDir)
```
Comment (dnr) [inline src/image_copy/main.go:115]:
```suggestion
	files, err := os.ReadDir(tmpDir)
```
ioutil is deprecated

File: src/image_copy/main.go (hunk 1 above)
Comment (stephanos) [reply src/image_copy/main.go:91]: Right; I would have to add that as a dependency. Do you feel strongly about this approach?

Comment (stephanos) [reply src/image_copy/main.go]: Okay, changed.

File: src/image_copy/main.go (hunk 7 above)
Comment (stephanos) [reply src/image_copy/main.go]: Oddly enough, Goland showed me a warning for the other functions but not this one.

File: src/image_copy/main.go (hunk 1 above)
Comment (dnr) [reply src/image_copy/main.go]: no big deal, fine either way

**Pull request merged**